src/Sentry/SentryFailedRequestHandler.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Sentry.Protocol;
+
+namespace Sentry
+{
+    internal class SentryFailedRequestHandler : ISentryFailedRequestHandler
+    {
+        private readonly IHub _hub;
+        private readonly SentryOptions? _options;
+
+        public string MechanismType { get => "SentryFailedRequestHandler"; }
+
+        /// <summary>
+        /// Initializes an instance of <see cref="SentryFailedRequestHandler"/>.
+        /// </summary>
+        internal SentryFailedRequestHandler(IHub hub, SentryOptions? options)
+        {
+            _hub = hub;
+            _options = options;
+        }
+
+        /// <summary>
+        /// Automaticlaly capture HTTP Client errors when captureFailedRequests
+        /// </summary>
+        /// <param name="request">The HttpRequestMessage sent</param>
+        /// <param name="response">The HttpResponse received</param>
+        public void CaptureEvent(HttpRequestMessage request, HttpResponseMessage response)
+        {
+            if (request is null)
+                return;
```

Comment:
The `request` will never be null.  But this would be a good place to do a sanity check:

```csharp
if (response.IsSuccessStatusCode)
{
    return;
}
```

Regardless of the range provided, we'd not want to do anything for success status.

Reply:
Also please use braces on all if-statement bodies throughout.  Thanks.